manifests/params.pp

```diff
@@ -195,9 +206,10 @@
   $server_http_allow          = []
 
   # use puppetserver (JVM) or puppet master (Ruby)?
-  $server_implementation = $aio_package ? {
-    true    => 'puppetserver',
-    default => 'master',
+  if $aio_package or ($::osfamily == 'Archlinux') {
+    $server_implementation = 'puppetserver'
```

Comment:
excellent point, will fix the remaining issues, run tests and squash all commits for clarity.